homeassistant/components/switchbot/sensor.py

```diff
@@ -0,0 +1,79 @@
+"""Support for SwitchBot sensors."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DATA_COORDINATOR, DOMAIN, MANUFACTURER, SensorType
+from .coordinator import SwitchbotDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbot sensor based on a config entry."""
+    coordinator: SwitchbotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    sensors = []
+
+    if coordinator.data[entry.unique_id].get("data"):
+        for sensor in coordinator.data[entry.unique_id].get("data"):
+            if sensor in SensorType.__members__:
```

Comment:
Use `EntityDescription` 

Examples:

https://github.com/home-assistant/core/pull/56395
https://github.com/home-assistant/core/pull/55932
https://github.com/home-assistant/core/pull/55218
https://github.com/home-assistant/core/pull/55931

Reply:
Thank you for all the examples, it helped a lot!

Done.